File: my_app/lib/models/product.dart
class Product {
	final String id;
	final String title;
	final String artist;
	final double price;
	final String genre;
	final String decade;
	final List<String> images;
	final String condition;
	final double rating;
	final String description;

	Product({
		required this.id,
		required this.title,
		required this.artist,
		required this.price,
		required this.genre,
		required this.decade,
		required this.images,
		required this.condition,
		required this.rating,
		required this.description,
	});

	Product.fromJson(Map<String, dynamic> json)
			: id = (json['_id'] ?? '').toString(),
				title = (json['title'] ?? '').toString(),
				artist = (json['artist'] ?? '').toString(),
				price = _asDouble(json['price']),
				genre = (json['genre'] ?? '').toString(),
				decade = (json['decade'] ?? '').toString(),
				images = (json['images'] as List?)
								?.map((image) => image.toString())
								.toList() ??
						<String>[],
				condition = (json['condition'] ?? '').toString(),
				rating = _asDouble(json['rating']),
				description = (json['description'] ?? '').toString();

	Map<String, dynamic> toJson() {
		return {
			'_id': id,
			'title': title,
			'artist': artist,
			'price': price,
			'genre': genre,
			'decade': decade,
			'images': images,
			'condition': condition,
			'rating': rating,
			'description': description,
		};
	}

	static double _asDouble(dynamic value) {
		if (value is num) {
			return value.toDouble();
		}

		if (value is String) {
			return double.tryParse(value) ?? 0.0;
		}

		return 0.0;
	}
}

File: backend/models/Product.js
const mongoose = require("mongoose");

const productSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: true,
      trim: true,
    },
    artist: {
      type: String,
      required: true,
      trim: true,
    },
    price: {
      type: Number,
      required: true,
    },
    genre: {
      type: String,
      required: true,
      enum: [
        "Rock",
        "Jazz",
        "Hip-Hop",
        "Electronic",
        "Classical",
        "Pop",
        "Alternative",
        "Indie",
        "Alternative Rock",
        "Indie Rock",
        "Pop Rock",
        "Indie Pop",
        "Synth-pop",
        "Art Pop",
        "Art Rock",
        "Classic Rock",
        "Psychedelic Rock",
        "Psychedelic Pop",
        "Experimental Rock",
        "Ambient Pop",
        "Dream Pop",
        "Slowcore",
        "Alternative R&B",
        "R&B",
        "Lo-fi",
        "Dark Pop",
        "Soft Rock",
        "Folk Pop",
        "Acoustic Pop",
        "EDM",
      ],
    },
    decade: {
      type: String,
      required: true,
      enum: ["1960s", "1970s", "1980s", "1990s", "2000s", "2010s", "2020s"],
    },
    images: [String],
    condition: {
      type: String,
      required: true,
      enum: ["Mint", "Near Mint", "Very Good", "Good", "Fair"],
    },
    rating: {
      type: Number,
      default: 0,
      min: 0,
      max: 5,
    },
    description: {
      type: String,
      default: "",
    },
  },
  { timestamps: true }
);

productSchema.index({ artist: "text", title: "text" });

module.exports = mongoose.model("Product", productSchema);
